### ISSUE TITLE
Link Container Check History with Part Attribute Sample Quality

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#1371]
In effort to track sample quality, create auto-update between container check history/condition and new part attribute tracking sample quality.

**Comment by dustymc** [comment]
"Auto" sounds scary.

I could add a "add {type something here} to {whatever field we're targeting with this thing} for all 275,374 specimen parts in this container" or etc. widget somewhere (edit container?).

**Comment by KyndallH** [comment]
I like auto - it sounds like it makes my life easier. I think auto in this instance means a link between check history/condition and new part attribute tracking sample quality so we don't have to enter the data in two spots.

**Comment by dustymc** [comment]
You (or your data loggers) update a freezer. "Auto" would create an attribute for all ~50K parts.

- is that what you're looking for?
- what would be the value of the auto-inserted attribute?
- what's that mean when a part was "rotten" when it went into the nitrogen, but ^^ inserts "excellent" based on whatever ya'll tell me?

**Comment by campmlc** [comment]
Is this scenario possible? Kyndall?
A freezer goes down and we record this as a Container Environment Record in Object Tracking (as a Temperature attribute, for example).  All the parts in the freezer would automatically be updated with a part attribute showing that temperature for that date.  This would be denormalized, but useful in seeing the history of each part to determine current condition.
The part could also have an attribute of "rotten"for an earlier date. There could be multiple part attributes related to condition. These are only visible when the part table is expanded?

Currently, we can go to Edit parts from the specimen record and click on history (an obscure pathway) to see the history of the part locations. But we can't access Container Environment from the specimen record. Container Environment would be much more useful if it could link to the part attributes to provide history and condition info. Or somehow add this into history?

The problem is that container environment affects part condition, but the data are in two separate places.

**Comment by dustymc** [comment]
>possible

Yes, but I'm still not (entirely, I think) convinced that it's a good idea. We've decided on pathway, have invested in it for a while at this point, and now we're looking at some weird hybrid that requires all the complexity of a normalized model but realizes few of the advantages??

We need to back up and analyze why the chosen model seems to not be sufficient even before it's been used. (Or maybe we've always seen this as sufficient in conjunction with containers??) An "add an attribute for a million parts" button may be a useful direction, but I think that situation should be approached very cautiously. Can you send me some of these sort of data so I can SQL them in and we can see what this looks like before we start adding magical/powerful/expensive buttons?

Also, what happens when a student clicks the button on the institution-container a few times and inadvertently adds millions of part attributes, including those parts which are in jars of ethanol and freezers inside the room?

>an obscure pathway

Which exists for obscure reasons. What/where/etc. would you prefer to see those data??

>But we can't access Container Environment from the specimen record.

I've been told ya'll want to use part attributes. I don't think we currently have funding to develop both, and there are many usability reasons to avoid doing one thing in two ways. 

>container environment affects part condition, but the data are in two separate places.

Maybe that is an argument for automation - just push all container environment updates to all specimens so everything's in one place and let users sort it out??? That would result in a lot of useless information - the {whatever someone records for a room} data probably don't have much effect on a newt in a jar of ethanol - but perhaps that's a price we should pay to keep everything accessible from the specimen (and tissue etc.) records.

That would likely eventually require adding infrastructure. We can cross that bridge when we get there, just something to keep in mind.

If we do go there, we should probably somehow be very explicit in the source of those data - "someone clicked the button [or updated, if we automate] on the container that contains the container that contains.....this specimen" vs. "someone looked at this specific part" (and a bunch of stuff in the middle).

>visible

That is probably another Issue. If ya'll use this, and especially if we automate it, "partDetail" (however those data are transmitted/displayed) is going to (potentially) get very large/complex, we're going to have to (eventually) do something which reflects that complexity for GGBN, etc. Who gets to see what, and where? I don't think any of that should ever drive a model and we don't need a final answer now, but it does need to remain something we're aware of.

>part table is expanded

That definitely deserves it's own Issue. All of that junk is there now because someone asked for it to be, and so everything will remain visible (and searchable) until I receive explicit instructions to do something else. Technically I don't think it's much of a problem, I just need a clear and documented instructions.

**Comment by Jegelewicz** [comment]
See also https://github.com/ArctosDB/arctos/issues/4161#issuecomment-988005603
